sdk/src/cose_sign.rs

```diff
@@ -155,31 +106,37 @@ pub async fn cose_sign_async(
     let alg = signer.alg();
 
     // build complete header
-    let (protected_header, unprotected_header) = build_headers(
-        data,
-        alg,
-        signer.certs()?,
-        signer.time_authority_url(),
-        signer.ocsp_val(),
-    )?;
+    let (protected_header, unprotected_header) = if _sync {
+        build_headers(signer, data, alg)?
+    } else {
+        build_headers_async(signer, data, alg).await?
+    };
 
-    let aad = b""; // no additional data required here
+    let aad: &[u8; 0] = b""; // no additional data required here
 
     let sign1_builder = CoseSign1Builder::new()
         .protected(protected_header)
         .unprotected(unprotected_header)
         .payload(data.to_vec());
 
+    let sign1_builder = if _sync {
+        sign1_builder.try_create_signature(aad, |bytes| signer.sign(bytes))?
+    } else {
+        sign1_builder
+    };
+
     let mut sign1 = sign1_builder.build();
 
-    let tbs = coset::sig_structure_data(
-        coset::SignatureContext::CoseSign1,
-        sign1.protected.clone(),
-        None,
-        aad,
-        sign1.payload.as_ref().unwrap_or(&vec![]),
-    );
-    sign1.signature = signer.sign(tbs).await?;
+    if _async {
+        let tbs = coset::sig_structure_data(
```

Comment:
@mauricefisher64 can this logic be coalesced with the block up at line 121 or does it need to be in this sequence?

Reply:
Yes, you can eliminate 122-126 and just sign the way we do with async buy explicitly signing the tbs.

Reply:
@mauricefisher64 thanks … done!